### ISSUE TITLE
dotnet failing to build on linux

### ISSUE THREAD

**Issue by parrt** [antlr/antlr4#1725]
Build errors still with mono/dotnet after merging [dotnet PR](https://github.com/antlr/antlr4/pull/1229). See [travis build](https://travis-ci.org/antlr/antlr4/jobs/207363550):

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
......Running [CSharp:APlus]
can't exec recognizer
java.lang.RuntimeException: Could not locate mono	at org.antlr.v4.test.runtime.csharp.BaseCSharpTest.locateTool(BaseCSharpTest.java:438)	at org.antlr.v4.test.runtime.csharp.BaseCSharpTest.getExecTestArgs(BaseCSharpTest.java:641)	at 
...
```

which is looking in 3 places:

```java
	private String locateTool(String tool) {
		String[] roots = { "/opt/local/bin/", "/usr/local/bin/", "/usr/bin/" };
		for(String root : roots) {
			if(new File(root + tool).exists())
				return root + tool;
		}
		throw new RuntimeException("Could not locate " + tool);
	}
```

It appears to be installing something:

```
Note, selecting 'dotnet-dev-1.0.0-preview2-1-003177' for regex 'dotnet-dev-1.0.0-preview2.1-003177'
The following extra packages will be installed:
  dotnet-host dotnet-hostfxr-1.1.0
  dotnet-sharedframework-microsoft.netcore.app-1.1.0 liblttng-ust-ctl2
  liblttng-ust0 libunwind8 liburcu1
The following NEW packages will be installed:
  dotnet-dev-1.0.0-preview2-1-003177 dotnet-host dotnet-hostfxr-1.1.0
  dotnet-sharedframework-microsoft.netcore.app-1.1.0 liblttng-ust-ctl2
  liblttng-ust0 libunwind8 liburcu1
0 upgraded, 8 newly installed, 0 to remove and 151 not upgraded.
...
```

Script is:

```bash
# install dotnet
sudo sh -c 'echo "deb [arch=amd64] https://apt-mo.trafficmanager.net/repos/dotnet-release/ trusty main" > /etc/apt/sources.list.d/dotnetdev.list'
sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 417A0893
sudo apt-get update
```

Travis script is:

```
    - os: linux
      jdk: oraclejdk7
      dist: trusty
      env: TARGET=dotnet
```

This might be related to a fix for #1407.

**Comment by parrt** [comment]
Ok, adding `mono --version` after the install gives:

```
./.travis/before-install-linux-dotnet.sh: line 22: mono: command not found
```

So the `before-install-linux-dotnet.sh` script doesn't install dotnet properly, right? It's not in the `PATH`.

Ter

**Comment by xied75** [comment]
@parrt Sorry about the slow response, main work got all time away. This error is that we tried to supply an variable to running code passing Maven, to switch the code logic to use "dotnet cli" rather than mono, without this switch everything still works as old mono target, the thing is then the test runner does not seem to be able to fetch the environment variable we provided, so it defaults to mono.

I'll try to have a look of this tonight. But don't waste more time on figuring out anything to do with mono, dotnet is totally independent from mono.

It must be something merged before our PR changed the way -DArgLines somehow.
